### PR TITLE
Feat/jest like

### PR DESCRIPTION
**Pull request by streamich**
1. adds `npm run demo` to see sample output
  2. Pretty prints the output similar to how Jest does it but still keeps the valid TAP output:

Things for discussion:

  1. Should `jest-tap-reporter` follow Jest convention of putting tests in `__tests__` folders?
  2. Now errors are logged into `STRERR` but TAP harness is supposed to ignore `STRERR` and 
  read only from `STDOUT` should. Should we log errors to `STDOUT`? Or into both file
  descriptors?

TODO:
  1. [ ] Implement `Ran all test suites matching /test\/TapReporter.spec.js/i.`
  2. [ ] Show error when it happens in suite or mocks and test suite does not even get a chance to run.
  3. [ ] When there are unused snapshots Jest exits with error, but we currently don't show any error.

### CODE REVIEW

File: src/TapReporter.js [hunk 1]
```diff
@@ -1,98 +1,121 @@
...
+    case STATUS_FAILED:
+      this.writer.failed(formattedTitle);
+      this.writer.errors(failureMessages);
+      break;
+    case STATUS_PENDING:
+      this.writer.pending(formattedTitle);
+      this.writer.errors(failureMessages);
+      break;
+    default:
+      this.commentLight(chalk`{italic Unknown status: ${status}}`);
```
Comment (carpasse) [inline src/TapReporter.js]: `this.writter.commentLight`??

File: src/TapReporter.js [hunk 2]
```diff
@@ -1,98 +1,121 @@
...
-    this._shouldFail = numFailedTestSuites > 0 || numFailedTests > 0;
-
-    this.logger.info('\n');
-    if (numFailedTestSuites > 0) {
-      this.logger.info(`# testSuites: ${skippedTestSuites}${chalk.red(`${numFailedTestSuites} failed`)}, ${numTotalTestSuites} total`);
-    } else {
-      this.logger.info(`# testSuites: ${skippedTestSuites}${chalk.green(`${numPassedTestSuites} passed`)}, ${numTotalTestSuites} total`);
-    }
+  onRunStart (results, options) {
+    this.onRunStartResults = results;
```
Comment (carpasse) [inline src/TapReporter.js]: `this.onRunStartResults` doesn't seem to be used.

File: test/index.spec.js [hunk 3]
```diff
@@ -4,3 +4,6 @@ const TapReport = require('../src/TapReporter');
 test('index must export TapReport', () => {
   expect(index).toBe(TapReport);
 });
+
+// eslint-disable-next-line jest/no-disabled-tests
+xit('a sample todo test', () => {});
```
Comment (carpasse) [inline test/index.spec.js]: should this test be on the demo??

File: src/TapReporter.js [hunk 4]
```diff
@@ -1,98 +1,121 @@
...
-      logLevel
-    });
-    this.counter = 0;
-
-    this.logger.log('\n');
-    this.logger.info('\n\n# Starting ...\n');
+    this.globalConfig = globalConfig;
+    this.options = options;
+    this[sShouldFail] = false;
+    this.logger = new Logger({logLevel});
```
Comment (carpasse) [inline src/TapReporter.js]: this.logger is not used outside of the writer. Can we make it private>

File: src/TapReporter.js (hunk 1 above)
Comment (streamich) [reply src/TapReporter.js]: good spot 👍

File: test/index.spec.js (hunk 3 above)
Comment (streamich) [reply test/index.spec.js]: yes, moved

File: src/TapReporter.js (hunk 1 above)
Comment (streamich) [reply src/TapReporter.js]: Should handle that default case better, and test it.

File: src/TapReporter.js (hunk 2 above)
Comment (streamich) [reply src/TapReporter.js]: We might use it in the future, now using it and removed it.

File: src/TapReporter.js (hunk 4 above)
Comment (streamich) [reply src/TapReporter.js]: Removed it instead, because not used.

**Pull request merged**